snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,263 @@
+
+        wheels = []
+        with tempfile.TemporaryDirectory() as temp_dir:
+
+            # Using pip with a few special parameters:
+            #
+            # --no-index: Don't hit pypi, assume the packages are already
+            #             downloaded (i.e. by using `self.download()`)
+            # --find-links: Provide the directory into which the packages
+            #               should have already been fetched
+            # --wheel-dir: Build wheels into a temporary working area rather
+            #              rather than cwd. We'll copy them over. FIXME: Why
+            #              not just build them straight in the package_dir?
+            self._run(['wheel', '--no-index', '--find-links',
+                       self._python_package_dir, '--wheel-dir',
+                       temp_dir] + list(args), **kwargs)
+            wheels = os.listdir(temp_dir)
+            for wheel in wheels:
+                file_utils.link_or_copy(
+                    os.path.join(temp_dir, wheel),
+                    os.path.join(self._python_package_dir, wheel))
+
+        return [os.path.join(self._python_package_dir, wheel)
+                for wheel in wheels]
+
+    def list(self):
+        """Determine which packages have been installed.
+
+        :return: Dict of installed python packages and their versions
+        :rtype: dict
+        """
+        output = self._run(['list', '--format=json'])
+        packages = collections.OrderedDict()
+        try:
+            json_output = json.loads(
+                    output, object_pairs_hook=collections.OrderedDict)
+        except json.decoder.JSONDecodeError as e:
+            raise errors.PipListInvalidJsonError(output) from e
+
+        for package in json_output:
+            if 'name' not in package:
+                raise errors.PipListMissingFieldError('name', output)
+            if 'version' not in package:
+                raise errors.PipListMissingFieldError('version', output)
+            packages[package['name']] = package['version']
+        return packages
+
+    def clean_packages(self):
+        """Remove the package cache."""
+        with contextlib.suppress(FileNotFoundError):
+            shutil.rmtree(self._python_package_dir)
+
+    def env(self):
+        """The environment used by pip.
+
+        This function is only useful if you happen to need to call into pip's
+        environment without using the API otherwise made available here (e.g.
+        calling the setup.py directly instead of with pip).
+
+        :return: Dict of the environment necessary to use the pip contained
+                 here.
+        :rtype: dict
+        """
+        env = os.environ.copy()
+        env['PYTHONUSERBASE'] = self._install_dir
+        if self._python_command.startswith(self._stage_dir):
+            env['PYTHONHOME'] = os.path.join(self._stage_dir, 'usr')
+        else:
+            env['PYTHONHOME'] = os.path.join(self._install_dir, 'usr')
```

Comment:
what about adding get_python_home to python_finder?

Reply:
+1

Reply:
Good call, done.